javascript/ql/lib/semmle/javascript/frameworks/Spife.qll

```diff
@@ -303,114 +288,128 @@ module Spife {
+    TemplateCall() { this.getCalleeName() = "template" }
 
-    TemplateInput() {
-      obj.getALocalSource().(DataFlow::ObjectLiteralNode).hasPropertyWrite(_, this)
-    }
+    override DataFlow::SourceNode getOutput() { result = this }
 
-    override RouteHandler getRouteHandler() { result = obj.getRouteHandler() }
+    override DataFlow::Node getTemplateFileNode() { result = this.(ReplyCall).getArgument(0) }
+
+    override DataFlow::Node getTemplateParamsNode() { result = this.(ReplyCall).getArgument(1) }
   }
 
   /**
    * An object passed to the `template` method of the reply object.
    */
   private class TemplateObjectInput extends DataFlow::Node {
-    ReplySource reply;
+    TemplateCall call;
 
-    TemplateObjectInput() {
-      exists(DataFlow::MethodCallNode call |
-        reply.ref() = call and
-        call.getMethodName() = "template" and
-        this = call.getArgument(1)
-      )
-    }
+    TemplateObjectInput() { this = call.(ReplyCall).getArgument(1) }
 
     /**
      * Gets the route handler that uses this object.
      */
-    RouteHandler getRouteHandler() { result = reply.getRouteHandler() }
+    RouteHandler getRouteHandler() { result = call.(ReplyCall).getRouteHandler() }
   }
 
   /**
-   * An invocation of the `redirect` method of an HTTP response object.
+   * An expression passed to the `template` method of the reply object
+   * as the value of a template variable.
    */
-  private class RedirectInvocation extends Http::RedirectInvocation, DataFlow::MethodCallNode instanceof ReplySource {
-    RedirectInvocation() { this.ref().(DataFlow::MethodCallNode).getMethodName() = "redirect" }
+  private class TemplateInput extends Http::ResponseBody {
+    TemplateObjectInput obj;
 
-    override DataFlow::Node getUrlArgument() { result = this.getAnArgument() }
+    TemplateInput() { obj.getALocalSource().hasPropertyWrite(_, this) }
 
-    override RouteHandler getRouteHandler() { result = this.getRouteHandler() }
+    override RouteHandler getRouteHandler() { result = obj.getRouteHandler() }
   }
 
   /**
-   * A call to `reply.template('template', { ... })`, seen as a template instantiation.
+   * An invocation of the `redirect` method of an HTTP response object.
    */
-  private class TemplateCall extends Templating::TemplateInstantiation::Range,
-    DataFlow::MethodCallNode instanceof ReplySource {
-    TemplateCall() { this.getMethodName() = "template" }
+  private class RedirectInvocation extends Http::RedirectInvocation instanceof ReplyCall {
+    RedirectInvocation() { this.getCalleeName() = "redirect" }
 
-    override DataFlow::SourceNode getOutput() { result = this }
-
-    override DataFlow::Node getTemplateFileNode() { result = this.getArgument(0) }
+    override DataFlow::Node getUrlArgument() { result = this.getAnArgument() }
 
-    override DataFlow::Node getTemplateParamsNode() { result = this.getArgument(1) }
+    override RouteHandler getRouteHandler() { result = this.getRouteHandler() }
```

Comment:
You are defining `getRouteHandler` as `getRouteHandler`? 

I think you are missing a super.  
```suggestion
    override RouteHandler getRouteHandler() { result = super.getRouteHandler() }
```